src/main/scala/com/raphtory/spouts/FileSpout.scala

```diff
@@ -86,6 +87,7 @@ class FileSpout[T: TypeTag](val path: String = "", val lineConverter: (String =>
 
   private def processFile(file: File) = {
     logger.info(s"Spout: Processing file '${file.toPath.getFileName}' ...")
+    SpoutTelemetry.totalFilesProcessed.inc()
```

Comment:
is there anyway we can adjust this slightly, the counter should only increment after the getlines? Or it can increment where you have put it, but then decrement if the spout throws an error. 
Can we then add an error case, where we increment how many file errors the spout throws?

Reply:
This would make the iterator not work. I've instead just placed it before calling `getLines`, we are also tracking errors in processed files now so we should be good to calculate.